.devcontainer/build.sh

```diff
@@ -74,4 +76,4 @@ CmakeBuild() {
   fi
 }
 
-[[ $SOURCED == "false" ]] && main "$@" || echo "Sourced!"
\ No newline at end of file
+[[ $SOURCED == "false" ]] && main "$@" || echo "Sourced!"
```

Comment:
for the build `$SOURCED` will be "false". Then when `main "$@"` returns a non 0 exit status (build failed), then the `||` part will be triggered and instead of an error the `echo "Sourced!"` will be executed. At least that's how I read this statement

I think we should split this statement into two lines:

```suggestion
[[ $SOURCED == "false" ]] && main "$@"
[[ $SOURCED != "false" ]] && echo "Sourced!"
```

same change should be done for `docker/build.sh`

Reply:
The original line is correct. For some reason the shell interprets `command && command || command` as an if-else. You can test this by running `true && echo true || echo false` and `false && echo true || echo false` in your terminal.

Reply:
The statement `true && false || echo false` prints `false` on the command line
```

Reply:
I think we should replace it with an explicit if/else if that is the intent of this line. Just to prevent confusion in the future

```suggestion
if [[ $SOURCED == "false" ]]; then
  main "$@"
else
  echo "Sourced!"
fi
```

Reply:
I'm witth @NeroBurner on this one. `true && false || echo false` prints `false` in my bash as well.
An explicit if else seems to be more clear in any way.